examples/llm_compression/torch/qat_with_lora/main.py

```diff
@@ -0,0 +1,418 @@
+    )
+    parser.add_argument("--epochs", type=int, default=10, help="Number of epochs.")
+    parser.add_argument("--batch_size", type=int, default=32, help="Size of training batch.")
+    parser.add_argument(
+        "--microbatch_size",
+        type=int,
+        default=8,
+        help="Size of each training microbatch. Gradients will be accumulated until the batch size is reached.",
+    )
+    return parser
+
+
+def main(argv) -> float:
+    """
+    Fine-tunes the specified model and returns the best validation similarity score.
+    """
+    parser = get_argument_parser()
+    args = parser.parse_args(argv)
+    assert torch.cuda.is_available()
+    set_seed(42)
+    device = "cuda"
+    torch_dtype = torch.bfloat16
+    compression_config = dict(
+        mode=CompressWeightsMode.INT4_ASYM, group_size=64, compression_format=CompressionFormat.FQ_LORA
+    )
+
+    # Configure output and log files.
+    output_dir = Path(args.output_dir)
+    tensorboard_dir = output_dir / "tb" / datetime.now().strftime("%D-%T")
+    last_dir = output_dir / "last"
+    best_dir = output_dir / "best"
+    for path in [output_dir, tensorboard_dir, last_dir, best_dir]:
+        if not args.resume:
+            shutil.rmtree(path, ignore_errors=True)
+        path.mkdir(exist_ok=True, parents=True)
+    wwb_ref_file = output_dir / "wwb_ref.csv"
+    ckpt_file = last_dir / "nncf_checkpoint.pth"
+    print(f"To visualize the loss and validation metrics, open Tensorboard using the logs from: {tensorboard_dir}")
+    tb = SummaryWriter(tensorboard_dir, "QAT with absorbable LoRA")
+
+    # Load original model and tokenizer.
+    model = AutoModelForCausalLM.from_pretrained(args.pretrained, torch_dtype=torch_dtype, device_map="auto")
+    tokenizer = AutoTokenizer.from_pretrained(args.pretrained)
+
+    # Use WhoWhatBench tool (WWB) is for validation during tuning. It estimates the similarity score between embedding
+    # computed by for data generated by two models, original floating-point one and optimized.
+    # TODO: (nlyalyus) Use original model for collecting reference, once the bug in WWB resolved.
+    wwb_ref_model = AutoModelForCausalLM.from_pretrained(args.pretrained, torch_dtype=torch_dtype, device_map="cpu")
+    save_wwb_ref(wwb_ref_model, tokenizer, wwb_ref_file)
+    del wwb_ref_model
+
+    # Prepare training data and pre-compute hiddens of teacher model for distillation loss.
+    train_loader = get_wikitext2(nsamples=args.nsamples, seqlen=args.seqlen, tokenizer=tokenizer, device=device)
+    orig_hiddens = calc_hiddens(model, train_loader)
+
+    # Create or load model to tune with Fake Quantizers and absorbable LoRA adapters.
+    example_input = get_model_input(train_loader[0])
+    if args.resume and ckpt_file.exists():
+        model = load_checkpoint(model, example_input, ckpt_file)
+    else:
+        model = compress_weights(model, dataset=Dataset([example_input]), **compression_config)
```

Comment:
Use `nncf.compress_weights`, to be alignment with usage documentation

Reply:
Where did you find this documentation? :)

https://docs.openvino.ai/2023.3/weight_compression.html
![image](https://github.com/user-attachments/assets/ab3c9727-4a2a-44dc-89e1-1c88b69c0cb0)
https://github.com/openvinotoolkit/nncf/blob/develop/docs/usage/post_training_compression/weights_compression/Usage.md
![image](https://github.com/user-attachments/assets/87969071-3063-491b-bd0e-6ee39d15b874)